src/main/java/org/wildfly/security/credential/store/impl/KeyStoreCredentialStore.java

```diff
@@ -810,6 +815,23 @@ private void load(String type) throws CredentialStoreException {
         }
     }
 
+    private KeyStore getKeyStoreInstance(String type) throws CredentialStoreException {
+        if (providers != null) {
+            for (Provider p: providers) {
+                try {
+                    return KeyStore.getInstance(type, p);
+                } catch (KeyStoreException e) {
```

Comment:
I am going to go ahead and include this PR but I think elsewhere we have some code that checks the capabiltiies of the Provider in advance to find the compatible one - may be worth looking for for here so we are not relying on error handling.